transmart-rest-api/grails-app/controllers/org/transmartproject/rest/AbstractQueryController.groovy

```diff
@@ -82,12 +88,12 @@ abstract class AbstractQueryController implements Controller {
                 }
             }
         }
-        return params.collectEntries { String k, v ->
+        return (params as Map<String, Object>).collectEntries { String k, v ->
             if (!RequestUtils.GLOBAL_PARAMS.contains(k)) {
                 if(v instanceof Object[] || v instanceof List) {
-                    [k, v.collect { URLDecoder.decode(it, 'UTF-8') }]
+                    [k, v.collect { URLDecoder.decode(it.toString(), 'UTF-8') }]
                 } else {
-                    [k, URLDecoder.decode(v, 'UTF-8')]
+                    [k, URLDecoder.decode(v.toString(), 'UTF-8')]
```

Comment:
This change broke parsing of e.g. constraints. This still works for e2e, but not when testing with GB or Postman (any POST call that has constraint in parameters).

Reply:
That is weird, because that part should not be used when doing a POST, because of the `if(request.method == 'POST')` on line 81.

Reply:
ah sorry, I meant the changes to `getGetOrPostParams`, but I'm not sure which line exactly (just added the comment in a wrong line).

Reply:
@ewelinagr That problem should now be fixed.

Reply:
I can confirm, everything works great now